### PR TITLE
Build and publish binary releases

### LINKED ISSUE #216: publish binary releases

### PR DESCRIPTION
**Pull request by casey**
Fixes #216

### CODE REVIEW

File: .github/workflows/package.yaml [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+        TARGET: ${{ matrix.target }}
+        REF: ${{ github.ref }}
+        OS: ${{ matrix.os }}
+        TARGET_RUSTFLAGS: ${{ matrix.target_rustflags }}
+      run: ./bin/package
+
+    - name: Release Check
+      id: release
+      run: |
+        if [[ '${{ github.ref }}' =~ '^refs/tags/[[:digit:]]+[.][[:digit:]]+[.][[:digit:]]+$' ]]; then
```
Comment (terror) [inline .github/workflows/package.yaml:106]:
```suggestion
        if [[ '${{ github.ref }}' =~ ^refs/tags/[[:digit:]]+[.][[:digit:]]+[.][[:digit:]]+$ ]]; then
```
Do we need quotes here?

File: .github/workflows/package.yaml [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+          target_rustflags: '--codegen target-feature=+crt-static'
+
+    runs-on: ${{ matrix.os }}
+
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: Make openssl-src Use Strawberry Perl
+      if: matrix.os == 'windows-latest'
+      run: echo OPENSSL_SRC_PERL=C:/Strawberry/perl/bin/perl >> $GITHUB_ENV
```
Comment (terror) [inline .github/workflows/package.yaml:59]:
```suggestion
      run: echo OPENSSL_SRC_PERL=C:/Strawberry/perl/bin/perl >> "$GITHUB_ENV"
```

**Comment by terror** [comment]
Other than that, the battle-tested shell-script lgtm

File: .github/workflows/package.yaml (hunk 1 above)
Comment (casey) [reply .github/workflows/package.yaml:106]: Good catch! This works on zsh but not on bash.

File: .github/workflows/package.yaml (hunk 2 above)
Comment (casey) [reply .github/workflows/package.yaml]: Nice

**Pull request merged**